cpp/src/arrow/ipc/json-test.cc

```diff
@@ -81,6 +86,11 @@ void TestArrayRoundTrip(const Array& array) {
   ASSERT_OK(WriteArray(name, array, &writer));
 
   std::string array_as_json = sb.GetString();
+  // RapidJSON can use SIMD operations that read past string
+  // but make ASAN complain, pad to make sure there is enough
+  // memory to keep ASAN happy (and accomodate if rapid json upgrades
+  // to SIMD instructions with 512 byte width).
+  array_as_json.reserve(array_as_json.size() + 512);
```

Comment:
Looks like this could be factored out.

Reply:
Used the other api, so I don't think its needed any longer.